pkg/docker/concurrent.go

```diff
@@ -0,0 +1,137 @@
+package docker
+
+import (
+	"context"
+	"fmt"
+	"sync"
+)
+
+type ConcurrentImageProcessor struct {
+	maxRoutines int
+}
+
+func NewConcurrentImageProcessor(maxRoutines int) *ConcurrentImageProcessor {
+	return &ConcurrentImageProcessor{maxRoutines: maxRoutines}
+}
+
+type ImageProcessor func(ctx context.Context, image string) error
+
+func (c *ConcurrentImageProcessor) Process(ctx context.Context, images []string, process ImageProcessor) error {
+	contextWithCancel, abortRemainingJobs := context.WithCancel(ctx)
+	workers := min(len(images), c.maxRoutines)
+	workers = max(workers, 1)
```

Comment:
isn't this always return 0 or 1?

Reply:
it will never return 0, that was the goal.
`workers` has to be at least one, never more than `maxRoutines` and never more than `len(images)`